dfastmi/io/RiverParameterData.py

```diff
@@ -0,0 +1,674 @@
+            A list of lists. Each list contains per reach within the corresponding
+            branch a list of 4 floats.
+        """
+        vals: Tuple[float, ...]
+        self.is_initialized()        
+        try:
+            g_val = self._config["General"][key]
+        except:
+            g_val = ""
+        all_values = []
+        for ib in range(len(self._branches)):
+            branch = self._branches[ib]
+            try:
+                b_val = self._config[branch][key]
+            except:
+                b_val = g_val
+            values_per_branch = []
+            for i in range(self._nreaches[ib]):
+                stri = str(i + 1)
+                try:
+                    val = self._config[branch][key + stri]
+                except:
+                    val = b_val
+                if val == "" and default is not None:
+                    vals = default
+                else:
+                    try:
+                        vals = tuple(float(x) for x in val.split())
+                    except:
+                        vals = ()
+                    if len(vals) != 4:
+                        raise Exception(
+                            'Reading {} for reach {} on {} returns "{}". Expecting {} values.'.format(
+                                key, i+1, self._branches[ib], val, 4
+                            )
+                        )
+                values_per_branch.append((vals[0], vals[1], vals[2], vals[3]))
+            all_values.append(values_per_branch)
+        return all_values
+
+    def collect_valuesN(
+        self,
+        key: str
+    ) -> List[List[Tuple[float, ...]]]:
+        """
+        This routines collects entries of type Tuple[float, ...]
+
+        Arguments
+        ---------
+        key : str
+            The name of the parameter for which the values are to be retrieved.
+
+        Returns
+        -------
+        data : List[List[Tuple[float, ...]]]
+            A list of lists. Each list contains per reach within the corresponding
+            branch a list of floats.
+        """
+        vals: Tuple[float, ...]
+        self.is_initialized()        
+        try:
+            g_val = self._config["General"][key]
+        except:
+            g_val = ""
+        all_values = []
+        for ib in range(len(self._branches)):
+            branch = self._branches[ib]
+            try:
+                b_val = self._config[branch][key]
+            except:
+                b_val = g_val
+            values_per_branch = []
+            for i in range(self._nreaches[ib]):
+                stri = str(i + 1)
+                try:
+                    val = self._config[branch][key + stri]
+                except:
+                    val = b_val
+                vals = tuple(float(x) for x in val.split())
+                values_per_branch.append(vals)
+            all_values.append(values_per_branch)
+        return all_values
+    
+    def config_get_bool(
```

Comment:
is this the correct method name? config makes it sound like it belongs in the Config class.

Reply:
Refactoring & improvements should be done in separate issue for each component

Reply:
https://issuetracker.deltares.nl/browse/DFAST-112